setup.py

```diff
@@ -3,23 +3,18 @@
 from io import open
 import os
 import sys
-
+from setuptools import Extension, setup
+from Cython.Build import cythonize
 
 here = os.path.abspath(os.path.dirname(__file__))
 
 # Get the long description from the README file
 with open(os.path.join(here, 'README.md'), encoding='utf-8') as f:
     long_description = f.read()
 
-cmake_args = []
-for option in ("CYTHON_LINETRACE", "CYTHON_ANNOTATE"):
-    value = os.environ.get(option)
-    if value is not None and bool(int(value)):
-        cmake_args.append("-D{}:BOOL=TRUE".format(option))
-
-# On Mac, require OSX >= 10.9 so we can use te new libc++ targeting C++11
-if sys.platform == "darwin":
-    cmake_args.append("-DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=10.9")
+extra_args = []
+if bool(int(os.environ.get('CYTHON_TRACE_NOGIL', '0'))):
+    extra_args.append(('CYTHON_TRACE_NOGIL', '1'))
```

Comment:
guess needs https://stackoverflow.com/a/28301932 also

Reply:
Two environment variable need to be supported by setup.py: CYTHON_ANNOTATE and CYTHON_LINETRACE.

they are used here by the `tox` test runner to enable code coverage of cython files:
https://github.com/harfbuzz/uharfbuzz/blob/bcebfee9f659309ef1a1074ad090d41b86a03688/tox.ini#L16-L17

CYTHON_LINETRACE should, at the same time, pass linetrace=True in cythonize call, and define the CYTHON_TRACE_NOGIL=1 macro.

Reply:
Thank you very much! `linetrace=True` results in failure so went with https://stackoverflow.com/a/28301932 solution.

Now I should learn how I can actually test this :)

Reply:
I don't like that we override cython global defaults.. passing linetrace=True ought to work. What failure did you get?

Reply:
Applying
```
diff --git a/setup.py b/setup.py
index d83e02b..052a9a6 100755
--- a/setup.py
+++ b/setup.py
@@ -48,7 +48,7 @@ setup_params = dict(
     zip_safe=False,
     setup_requires=["setuptools_scm"],
     python_requires=">=3.5",
-    ext_modules = cythonize(extension, annotate=bool(int(os.environ.get('CYTHON_ANNOTATE', '0'))))
+    ext_modules = cythonize(extension, linetrace=True, annotate=bool(int(os.environ.get('CYTHON_ANNOTATE', '0'))))
 )
```

getting
```
Traceback (most recent call last):
  File "./setup.py", line 51, in <module>
    ext_modules = cythonize(extension, linetrace=True, annotate=bool(int(os.environ.get('CYTHON_ANNOTATE', '0'))))
  File "~/.local/lib/python3.8/site-packages/Cython/Build/Dependencies.py", line 961, in cythonize
    c_options = CompilationOptions(**options)
  File "~/.local/lib/python3.8/site-packages/Cython/Compiler/Main.py", line 563, in __init__
    raise ValueError(message)
ValueError: got unknown compilation option, please remove: linetrace
```

Reply:
you need to pass `cythonize(..., compiler_directives={"linetrace": True})" as I do in skia-pathops here: 
https://github.com/fonttools/skia-pathops/blob/6581155e5151624ae9756392db025810933b74d6/setup.py#L108

Reply:
Nice, done